packages/nuxt/src/app/composables/ssr.ts

```diff
@@ -61,6 +61,43 @@ export function setResponseStatus (arg1: H3Event | number | undefined, arg2?: nu
   }
 }
 
+/** @since 3.14.0 */
+export function useResponseHeader (header: string) {
+  if (import.meta.client) {
+    if (import.meta.dev) {
+      return computed({
+        get: () => undefined,
+        set: () => console.warn('[nuxt] Setting response headers is not supported in the browser.'),
+      })
+    }
+    return ref()
+  }
+
+  const event = useRequestEvent()
+  if (!event) {
+    if (import.meta.dev) {
+      return computed({
+        get: () => undefined,
+        set: () => console.warn('[nuxt] Setting response headers is not supported in the browser.'),
```

Comment:
I think the warning text needs to be changed here? `import.meta.client` was already checked in the code above. In this scenario the `event` is missing

Reply:
you are quite right!